feature-libs/cart/quick-order/core/services/quick-order.service.ts

```diff
@@ -16,24 +16,44 @@ import {
   ProductSearchPage,
   SearchConfig,
 } from '@spartacus/core';
-import { BehaviorSubject, Observable, Subject } from 'rxjs';
+import {
+  BehaviorSubject,
+  Observable,
+  Subject,
+  Subscription,
+  timer,
+} from 'rxjs';
 import { filter, first, map, switchMap, take, tap } from 'rxjs/operators';
+import { ClearMessageTimout } from '../models/clear-message-timeout.model';
 
 @Injectable({
   providedIn: 'root',
 })
-export class QuickOrderService implements QuickOrderFacade {
+export class QuickOrderService implements QuickOrderFacade, OnDestroy {
   protected productAdded$: Subject<string> = new Subject<string>();
   protected entries$: BehaviorSubject<OrderEntry[]> = new BehaviorSubject<
     OrderEntry[]
   >([]);
+  protected deletedEntries$: BehaviorSubject<OrderEntry[]> =
+    new BehaviorSubject<OrderEntry[]>([]);
+  protected deletionClearTimeout = 5000;
+
+  private clearMessageTimeoutArray: ClearMessageTimout[] = [];
```

Comment:
I would not store these subscriptions in an array. It's easier to store them in an object where `productCode` would be the key and subscription would be the value. Easier to find particular element and to remove it.

I think the same applies to deltedEntries$. You make the operations (undo, clear) based on the `productCode`, so it should also be easier to find those in an object instead of array.